packages/sdk/src/modules/middlewareModule/types.ts

```diff
@@ -155,6 +155,31 @@ export type ErrorHandlerContext = {
   httpClient: HTTPClient;
 };
 
+/**
+ * Custom logger for the middlewareModule, allowing for request and response logging.
+ */
+export type Logger = {
+  onRequest?: (payload: {
```

Comment:
In many cases, if there is a type used like that and it is not exported, there will be a usecase where someone would like to use it, to create an object elsewhere and satisfy this type (for the type safety) or for other reason. I think that it will be no harm to export the payload type for a request and response.

Reply:
done [22c9082](https://github.com/vuestorefront/vue-storefront/pull/7227/commits/22c9082e868f2af96c4131d9c7e0ed97f83561c5)